src/Components/Files/AudioCaptureDialog.tsx

```diff
@@ -0,0 +1,203 @@
+import { useEffect, useState } from "react";
+import useRecorder from "../../Utils/useRecorder";
+import { Link } from "raviger";
+import CareIcon from "../../CAREUI/icons/CareIcon";
+import { useTimer } from "../../Utils/useTimer";
+import { t } from "i18next";
+
+export interface AudioCaptureDialogProps {
+  show: boolean;
+  onHide: () => void;
+  onCapture: (file: File) => void;
+  autoRecord?: boolean;
+}
+
+export default function AudioCaptureDialog(props: AudioCaptureDialogProps) {
+  type Status =
+    | "RECORDING"
+    | "WAITING_TO_RECORD"
+    | "PERMISSION_DENIED"
+    | "RECORDED";
+
+  const { show, onHide, onCapture, autoRecord = false } = props;
+
+  const [status, setStatus] = useState<Status | null>(null);
+
+  const [audioURL, , startRecording, stopRecording, , resetRecording] =
+    useRecorder((permission: boolean) => {
+      if (!permission) {
+        handleStopRecording();
+        resetRecording();
+        setStatus("PERMISSION_DENIED");
+      }
+    });
+
+  const timer = useTimer();
+
+  const handleStartRecording = () => {
+    setStatus("RECORDING");
+    startRecording();
+    timer.start();
+  };
+
+  const handleStopRecording = () => {
+    if (status !== "RECORDING") return;
+    setStatus("RECORDED");
+    stopRecording();
+    timer.stop();
+  };
+
+  const handleRestartRecording = () => {
+    if (status !== "RECORDED") return;
+    resetRecording();
+    handleStartRecording();
+  };
+
+  const handleSubmit = async () => {
+    const response = await fetch(audioURL);
+    const blob = await response.blob();
+    const file = new File(
+      [blob],
+      `recording_${new Date().toISOString().replaceAll(".", "_").replaceAll(":", "_")}.mp3`,
+      { type: "audio/mpeg" },
+    );
+    resetRecording();
+    onHide();
+    onCapture(file);
+  };
+
+  useEffect(() => {
+    const checkMicPermission = async () => {
+      try {
+        const permissions = await navigator.permissions.query({
+          name: "microphone" as PermissionName,
+        });
+        setStatus(
+          permissions.state === "denied"
+            ? "PERMISSION_DENIED"
+            : "WAITING_TO_RECORD",
+        );
+      } catch (error) {
+        setStatus(null);
+      }
+    };
+
+    show && checkMicPermission();
+
+    return () => {
+      setStatus(null);
+    };
+  }, [show]);
+
+  useEffect(() => {
+    if (autoRecord && show && status === "WAITING_TO_RECORD") {
+      handleStartRecording();
+    }
+  }, [autoRecord, status, show]);
+
+  return (
+    <div
+      className={`inset-0 bg-black/70 backdrop-blur transition-all ${show ? "visible opacity-100" : "invisible opacity-0"} fixed z-50 flex flex-col items-center justify-center gap-8 p-6 text-center`}
+    >
+      {status === "PERMISSION_DENIED" && (
+        <div>
+          <h2 className="font-bold text-white">
+            {t("audio.allow_permission")}
```

Comment:
Lets stick to snake casing, use double underscore (__) instead of a dot (.), please update the rest too